### PR TITLE
support implicit parameter list in algebras

### PR DESCRIPTION
**Pull request by fommil**
partial fix for #365 and a failing test for the remainder

### CODE REVIEW

File: freestyle/shared/src/test/scala/freestyle/free.scala [hunk 1]
```diff
@@ -129,6 +129,25 @@ class freeTests extends WordSpec with Matchers {
       "implicitly[SCtors1[SCtors1.Op]]" should compile 
     }
 
+    "allow implicit parameters in algebra members" in {
```
Comment (diesalbla) [inline freestyle/shared/src/test/scala/freestyle/free.scala:132]: Apart from these test cases, specific for type-classes, we should also add the following test cases:  

* A test case with a simpler implicit parameter: 
```Scala
   "@free trait A { def b(x: Int)(implicit y: Int): FS[Unit]" should compile
```

* A test case with only implicit and no non-implicit parameters: 
```Scala
"@free trait A { def b(implicit y: Int): FS[Unit]" should compile
```

**Review by diesalbla — APPROVED** [review]
Nice improvement.  I just left a small comment.
Thank you for your contribution, @fommil.

File: freestyle/shared/src/test/scala/freestyle/free.scala (hunk 1 above)
Comment (fommil) [reply freestyle/shared/src/test/scala/freestyle/free.scala:132]: ok, but you've caught me literally as I was about to go to bed so maybe somebody could add these tomorrow? :smile:

**Comment by fommil** [comment]
oh whoops, sorry I didn't update this. @raulraja is this still needed or did your recent PR fix it?

**Comment by raulraja** [comment]
Oh I missed this! I fixed it but did not know you could check for the implicit param the way you did it, so hold on until we fix an issue with CI and I'll add these commits with the extra tests and and your check in addition to what's in #377

**Comment by raulraja** [comment]
Additional tests were added on https://github.com/frees-io/freestyle/blob/08cd997a54ce3c617d4e2c6f48445ec8b43ed0d8/modules/core/shared/src/test/scala/freestyle/free.scala#L66-L95

**Pull request closed without merging**